src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts

```diff
@@ -176,43 +186,105 @@ export namespace NestedDropdownListHelper {
         return item;
     };
 
-    export const updateCategoryChecked = <V1, V2, V3>(
-        list: FormattedOptionMap<V1, V2, V3>,
+    export const getCategoryChecked = <V1, V2, V3>(
```

Comment:
what was the reason for switching from recursively updating to looping over the subitems?

the logic feels like it is repeated twice across `update` and the map iteration